hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpCommandParser.java

```diff
@@ -37,7 +39,16 @@ public final TextCommand parser(TextDecoder decoder, String cmd, int space) {
         } else {
             return new ErrorCommand(ERROR_CLIENT);
         }
-        return createHttpCommand(decoder, uri);
+        try {
+            String urlDecodedUri = decodeUrl(uri);
+            return createHttpCommand(decoder, urlDecodedUri);
+        } catch (UnsupportedEncodingException e) {
+            return new ErrorCommand(ERROR_CLIENT, "URL is not encoded properly");
```

Comment:
I believe this exception can be thrown if the encoding you passed to the `URLDecoder.decode` does not exist. https://docs.oracle.com/javase/8/docs/api/java/net/URLDecoder.html

So, the error message might be misleading.

Reply:
ok changed it.